joomla/joomla-cms
# Add a plugin with a nag message to update PHP

#### Summary of Changes

Far too many people run Joomla on outdated PHP versions (per [the stats](https://developer.joomla.org/about/stats.html) only 37% of Joomla! 3.5+ installs are on a currently supported PHP version).  To help alleviate this, I propose a nag message to tell people to upgrade.

This plugin checks three conditions: whether the PHP version is still fully supported, whether it has entered security support only, or whether it is unsupported by the PHP project.  I don't particularly care whether the user is running a PHP version custom compiled by some Linux distro that claims they are still supporting it; if PHP isn't supporting the branch anymore they need to update.
#### Testing Instructions

Apply the pull request; discover, install, and enable the plugin.  For PHP 5.3 thru 5.5 you should get a red error message telling you the PHP version is unsupported and you should upgrade yesterday.  PHP 5.6 and 7.0 won't emit anything because neither has reached a security only state, however if you want to test that aspect of things go into the plugin and change the date for when security support ends in the data array to a date before today (today being whatever day you test this).
#### Documentation Changes Required

None.  No new API is introduced.
#### Maintenance Efforts Added

As the support dates for PHP branches change (new versions added, dates postponed), the data array in this plugin will need to be updated.  Generally this is a once a year requirement when a new version is added only.
#### Joomla, Stop Nagging Me!!!

Upgrade, plain and simple.  Oh, not an option?  As long as you understand you are running unsupported software, you can disable the plugin and stop being nagged.  But you should still upgrade yesterday.
